### PR TITLE
Initial commit rate filter, fixes #177

### LINKED ISSUE #177: Rate limit requests on votes/classifications
I already opened #176 that addresses the duplicate classifications and #173 that fixes duplicate votes. We might however want to prevent bulk classification and votes to different songs or rooms. Therefore we should implement a rate limiter filter for some requests.

Marked as "Should have" because if the outstanding PR's will be merged, this issue does not influence the classification but only the awarded points.

### PR DESCRIPTION
**Pull request by jwgmeligmeyling**
Initial commit for review. Will add tests.

### CODE REVIEW

File: src/main/java/me/moodcat/api/filters/RateLimitFilter.java [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+@Singleton
+@javax.ws.rs.ext.Provider
+public class RateLimitFilter implements ContainerRequestFilter {
+
+    private static final int HTTP_TOO_MANY_REQUESTS = 429;
+
+    /**
+     * A map that temporary stores requests for a given method.
+     */
+    private final Multimap<Method, Entry> entries;
```
Comment (JeremybellEU) [inline src/main/java/me/moodcat/api/filters/RateLimitFilter.java]: When we have a lot of users (let's see more than one thousand), couldn't this become a problem? What are the limits to storing this in a map?

Comment (JeremybellEU) [inline src/main/java/me/moodcat/api/filters/RateLimitFilter.java]: Also we have a lot of synchronized blocks. There is no concurrent map that can serve the same purpose?

Comment (JWGmeligMeyling) [inline src/main/java/me/moodcat/api/filters/RateLimitFilter.java]: Theres no concurrent map for multimaps. :sad:

Comment (JWGmeligMeyling) [inline src/main/java/me/moodcat/api/filters/RateLimitFilter.java]: No this is not a problem because we do a clean up and the objects are really small.

**Comment by JeremybellEU** [comment]
Closing since we are discontinueing this project.

**Pull request closed without merging**